pkg/autoscaler/metrics/http_scrape_client.go

```diff
@@ -65,9 +76,20 @@ func (c *httpScrapeClient) Scrape(url string) (Stat, error) {
 	return statFromPrometheus(resp.Body)
 }
 
+func ioCopy(r io.Reader) ([]byte, error) {
+	buffer := pool.Get().(*bytes.Buffer)
+	buffer.Reset()
+	defer pool.Put(buffer)
```

Comment:
I don't think this is safe. We can only return the buffer after we've unmarshalled the stat. The underlying byte array will still be needed after this function.

I'd recommend to back out the bufferpool stuff from this PR honestly, it unnecessarily complicates things.

Reply:
Ok I will remove but it could be fixed easily by not calling the function,and move its contents in the statFromProto fucntion.

Reply:
sure but why not land the additional improvement in a followup PR?

Reply:
Yeah, also [it's not really safe](https://github.com/golang/go/issues/23199) to store a bytes.Buffer in a sync.Pool. 

Agree with @markusthoemmes this is overkill in this PR anyway, let's look at it as a follow-up: I mostly just wanted a Benchmark so we were sure this actually ends up being an improvement with the extra allocations.

Reply:
a@julz we could return an array of fixed size if I read correctly to solve this, in addition the writes here are small. Anyway I removed that part.